lib/pact_broker/client/tasks/publication_task.rb

```diff
@@ -87,6 +92,14 @@ def the_branch
           branch
         end
       end
+
```

Comment:
There was some complex logic I put in for autodetecting the branch where possible but not erroring if it wasn't there, that I don't think is necessary for the consumer version. The consumer version must be there - there's no way to publish it without one.

Reply:
Right this should be alot simpler now

- allow the user to provide own version, with auto_detect_version_properties true
- retrieve commit hash as version and no version supplied, with auto_detect_version_properties true
- error if no version supplied, with auto_detect_version_properties false (default behaviour)
- ensures user provides own version, with auto_detect_version_properties false (default behaviour)